src/components/ConfirmModal.tsx

```diff
@@ -27,14 +31,12 @@ export default function ConfirmModal({
       <Modal.Footer>
         <Button
           onClick={() => {
-            callBackFunc()
-              .then(() => {
-                closeModal();
-              })
-              .then(() => {
-                cleanUpFunc();
-              });
             closeModal();
+            callBackFunc().then((res: any) => {
+              if (res == expectedResult) {
+                cleanUpFunc();
+              } else return;
+            });
```

Comment:
This would benefit from some strong typing. Enforcing that the callBack returns a boolean or something would be useful.